File: apps/api-kotlin/src/main/kotlin/com/couplecalendar/application/dto/response/CoupleResponse.kt
package com.couplecalendar.application.dto.response

import com.couplecalendar.domain.aggregate.Couple

data class CoupleResponse(
    val id: String,
    val user1Id: String,
    val user2Id: String?,
    val startDate: String,
    val inviteCode: String?,
    val inviteCodeExpiresAt: String?,
    val daysFromStart: Long,
    val isComplete: Boolean,
    val createdAt: String,
    val updatedAt: String
) {
    companion object {
        fun fromAggregate(couple: Couple): CoupleResponse = CoupleResponse(
            id = couple.id.toString(),
            user1Id = couple.user1Id.toString(),
            user2Id = couple.user2Id?.toString(),
            startDate = couple.startDate.toString(),
            inviteCode = couple.inviteCode,
            inviteCodeExpiresAt = couple.inviteCodeExpiresAt?.toString(),
            daysFromStart = couple.getDaysFromStart(),
            isComplete = couple.isComplete(),
            createdAt = couple.createdAt.toString(),
            updatedAt = couple.updatedAt.toString()
        )
    }
}

data class InviteCodeResponse(
    val inviteCode: String,
    val expiresAt: String
)

File: apps/web/src/data/dto/couple-response.ts
export interface CoupleResponse {
	id: string;
	user1Id: string;
	user2Id: string | null;
	startDate: string;
	inviteCode: string | null;
	inviteCodeExpiresAt: string | null;
	daysFromStart: number;
	isComplete: boolean;
	createdAt: string;
	updatedAt: string;
}

export interface InviteCodeResponse {
	inviteCode: string;
	expiresAt: string;
}
